TextRight.Core/ObjectModel/Blocks/Text/TextCaretTextMinipulator.cs

```diff
@@ -8,42 +8,32 @@ namespace TextRight.Core.ObjectModel.Blocks.Text
   public static class TextCaretTextMinipulator
   {
     /// <summary> Deletes text at the given position. </summary>
-    public static TextCaret DeleteText(this TextCaret caret, int numberOfCharacters)
+    public static TextCaret DeleteText(this TextCaret caret, int numberOfGraphemes)
     {
       // TODO GRAPHEMES
 
-      //while (numberOfCharacters > 0)
-      {
-        int numberOfCharactersRemainingInCurrentFragment = caret.Span.NumberOfChars - caret.Offset.CharOffset;
-        int numberOfCharactersToRemove = numberOfCharacters;
-
-        if (numberOfCharactersToRemove > numberOfCharactersRemainingInCurrentFragment)
-        {
-          numberOfCharactersToRemove = numberOfCharactersRemainingInCurrentFragment;
-        }
-
-        // TODO special case when we're deleting the entire fragment
-        caret.Span.RemoveCharacters(caret.Offset.CharOffset, numberOfCharactersToRemove);
+      var endCaret = caret;
 
-        numberOfCharacters -= numberOfCharactersToRemove;
-        // TODO what happens for multiple fragments
-        return TextCaret.FromOffset(caret.Span, caret.Offset.GraphemeOffset);
+      while (numberOfGraphemes > 0)
+      {
+        var next = endCaret.GetNextPosition();
+        if (!next.IsValid)
+          break;
+        
+        endCaret = next;
+        numberOfGraphemes--;
       }
+      
+      // TODO special case when we're deleting the entire fragment
```

Comment:
Let's do this now

Reply:
It looks like this isn't needed now that we removed fragments - .NET already optimizes string deletions when deleting the whole string, so this is "done"